### PR TITLE
Refactoring `save-orchestrator-common`

### PR DESCRIPTION
**Pull request by nulls**
`save-agent` needs to know only url to where need to post heartbeats and result of run

### CODE REVIEW

File: save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt [hunk 1]
```diff
@@ -164,7 +158,7 @@ class SaveAgent(private val config: AgentConfiguration,
         while (true) {
             val response = runCatching {
                 // TODO: get execution progress here. However, with current implementation JSON report won't be valid until all tests are finished.
-                sendHeartbeat(ExecutionProgress(executionId = executionId.value, percentCompletion = 0))
+                sendHeartbeat(ExecutionProgress(executionId = requiredEnv(AgentEnvName.EXECUTION_ID).toLong(), percentCompletion = 0))
```
Comment (nulls) [inline save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt:161]: a single place where execution id is in used

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt [hunk 2]
```diff
@@ -7,9 +7,8 @@ package com.saveourtool.save.agent.utils
 import com.saveourtool.save.agent.AgentConfiguration
 import com.saveourtool.save.agent.AgentEnvName
 import com.saveourtool.save.core.logging.logTrace
-import generated.SAVE_CORE_VERSION
 
-internal const val SAVE_CLI_EXECUTABLE_NAME = "save-$SAVE_CORE_VERSION-linuxX64.kexe"
+internal const val SAVE_CLI_EXECUTABLE_NAME = "save-linuxX64.kexe"
```
Comment (petertrr) [inline save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt:11]: Same question here. For me it seems useful to know which versions of all components have been used for a particular execution to make debugging easier.

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt [hunk 3]
```diff
@@ -55,7 +54,7 @@ fun main() {
     }
         .updateFromEnv()
     logType.set(if (config.debug) LogType.ALL else LogType.WARN)
-    logDebugCustom("Instantiating save-agent version $SAVE_CLOUD_VERSION with config $config")
+    logDebugCustom("Instantiating save-agent with config $config")
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt:57]: ok. will try to add it

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt (hunk 2 above)
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt:11]: here (in file name) version is not required

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt]: added as env variable

**Comment by akuleshov7** [comment]
You will have a lot of problems with conflicts resolving if you won't merge it, may be merge?

**Comment by nulls** [comment]
> You will have a lot of problems with conflicts resolving if you won't merge it, may be merge?

yes, I would like to merge. But I need an approval

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/AgentService.kt [hunk 4]
```diff
@@ -23,4 +23,10 @@ class AgentService(private val agentRepository: AgentRepository) {
      */
     internal fun deleteAgentByExecutionIds(executionIds: List<Long>) =
             agentRepository.deleteByExecutionIdIn(executionIds)
+
+    /**
+     * @param containerId
+     * @return ID of [Execution]
+     */
+    internal fun getExecutionId(containerId: String): Long? = agentRepository.findByContainerId(containerId)?.requiredId()
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/AgentService.kt:31]: need to refactor it to be ready to case when a single container can be used for different executions (long running agent)

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**